spec/support/matchers/have_multivalued_field.rb

```diff
@@ -29,13 +29,20 @@
       @options_missing = (@options - @option_values)
     end
 
-    @field && @multiple && (!label || @label_match) && @options_missing.empty?
+    @field &&
+      (@multiple || @single) &&
+      (!label || @label_match) &&
+      @options_missing.empty?
   end
 
   chain :on_model,   :model_class
   chain :and_label,  :label
   chain :with_label, :label
 
+  chain :as_single_valued do
+    @single = true
+  end
+
```

Comment:
I don't see where the failure message tells me if the matcher fails on this part of the expectation.  Should it?

Reply:
Looking at this again, the matcher doesn't currently confirm that single valued fields are actually single valued. We need an `xor` instead of `||`.